### DISCUSSION TITLE
How to disable CORS headers from downstream services?

### DISCUSSION THREAD

**Discussion by a510** [ThreeMammals/Ocelot#1802] [Q&A] [answered]
I handle CORS in the Gateway project and everything works fine, but some of the downstream services mistakenly add CORS headers to the downstream response (e.g. `allow-access-control-origin: *`) which breaks the CORS configuration in the Gateway project.
Is there a built-in feature for disabling CORS headers from downstream services ?

**Comment by ks1990cn** [comment]
Gateway can not do that. Infact no api can change configuration of any other API.

**Reply by a510** [reply]
I didn't mean to change other APIs. I mean to remove CORS headers from downstream responses and set its own headers before returning the HTTP response.
Note that Ocelot already add some headers to the responses.

**Comment by raman-m** [comment, marked as answer]
@a510 Hi Ahmad!
Welcome to Ocelot world!

> Is there a built-in feature for disabling CORS headers from downstream services ?

**No**! You have to write special [Delegating Handler](https://ocelot.readthedocs.io/en/latest/features/delegatinghandlers.html) or [Middleware overriding](https://ocelot.readthedocs.io/en/latest/features/middlewareinjection.html) module to process downstream response headers.

**Reply by a510** [reply]
Thanks for your reply. 
I used a Delegating Handler.

**Reply by raman-m** [reply]
Consider closing the discussion if answered.
